data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -194,10 +210,15 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                 if (terms.length == 2) {
                     value = terms[1];
                 } else {
-                    LOG.debug(String.format("Unsuccessful match: '%s'", terms[0]));
+                    LOG.debug("Unsuccessful match: '{}'", terms[0]);
                     value = keyValueProcessorConfig.getNonMatchValue();
                 }
 
+                if (defaultKeysMap.containsKey(key) && defaultKeysMap.containsValue(value)) {
```

Comment:
@shenkw1 , I think we don't need this conditional at all.

The basic logic for defaults should be that the key/values are used from the event. But, if there is no such key in the event, set it to a certain default.